draft-ietf-quic-transport.md

```diff
@@ -1427,6 +1431,63 @@ a single packet.
+
+ECN is validated by setting the ECT(0) or ECT(1) bit in the IP header, following
+the guidelines of {{!RFC8311}}. Clients and servers both mark the IP packet
+containing an Initial packet ({{packet-initial}}). If ECN is supported, the
+recipient of a marked packet will increase the corresponding counter and send an
+ACK_ECN frame.
+
+An endpoint uses the ACK_ECN frame to verify that the ECT markings were received
+by its peer. The endpoint counts the number of acknowledged QUIC packets that
+were sent in marked IP packets. If the counters in the ACK_ECN frame match that
+number, the path and peer do not remove ECN markings. The endpoint records this
+path as being ECN capable and continues to mark IP packets that it sends.
+
+If an endpoint receives an ACK frame, indicating that no markings were received,
+or the counter in the ACK_ECN frame do not match the number of QUIC packets that
+were sent in marked IP packets, then the path or peer remove ECN markings. The
+endpoint records this path as not being ECN capable and it ceases marking of
+packets.
+
+IP packets sent on a new path SHOULD be marked with ECT(0) or ECT(1) to verify
+that the new path supports ECN, see {{ecn-connection-migration}}.
+
+### Continuous Verification of ECN {#ecn-continuous-verification}
+
+If the ECN capability check was successful and the endpoint continues to send
+ECT marked packets then continuous verification is applied. This is to detect
+any cases when ECN field is bleached, that is, zeroed out by a network node,
+likely as the result of a routing changes since the ECN capability check.
+
+For each received ACK_ECN frame, the total number of newly acknowledged packets
+can be compared to the total increase in ECN counters. If the increase in ECN
+counters is less, then an ECN failure has occurred and ECN should be disabled.
+ECN is also disabled in case an ACK frame is received acknowledging any ECT sent
+packet.
+
+If the acknowledgements from the receiver are lost such that one or more packet
+are received by the receiver, but never acknowledged to the sender an
+insensitivity to bleaching will be created. In this situation the ECN counters
+reported will have increase, but the sender side total for acknowledged packets
+will not have increased. Thus, a number of bleached packets equal to the number
+of packets that failed to be acknowledged can be received before triggering the
+continuous verification. To address this issue the sender detect the case when
+the ECN counters grows more than number of acknowledged packets when a ACK_ECN
+frame is received. In such cases a new comparison point is created by storing
+the current number of totally acknowledged packets and latest ECN counters. Then
+comparison are done by subtracting these stored values from the respective
+counters prior to the comparison. Note that any out-of-order ACK_ECN frames
+can't be used for determining any loss of acknowledgements.
```

Comment:
This whole paragraph is really difficult to understand. "Comparison points" are not mentioned anywhere else, for starters.